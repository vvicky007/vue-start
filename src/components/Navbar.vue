<template>
  <header>
    <Collapsible>
      <nav id="nav">
        <h1>Feel Football</h1>

        <ul class="nav-list-items">
          <li>
            <router-link to="/">Home</router-link>
          </li>
          <li >
            <router-link to="/leagues" @click="ClickedLeagues($event)">Leagues</router-link>
          </li>
          <li v-if="!getAuthenticity()">
            <router-link to="/form" >Signin</router-link>
          </li>
           <li v-if="getAuthenticity()">
            <router-link to="/profile" v-if="getAuthenticity">Profile</router-link>
          </li>
        </ul>
        <button @click="logOut" >LogOut</button>
      </nav>
    </Collapsible>
  </header>
</template>
<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import Collapsible from "@/shared/collapsible.vue";
import { firebaseStore } from "@/store";
@Component({
  name: "Navbar",
  components: {
    Collapsible,
  },
})
export default class Navbar extends Vue {
  isAuthenticated = firebaseStore.user
  created(){
    firebaseStore.getUserOnChange()
  }
  getAuthenticity():boolean{
    this.isAuthenticated = firebaseStore.user
    return this.isAuthenticated != null
  }
  
  async logOut(){
    await firebaseStore.logOut()
    this.$router.push('/')
  }
  ClickedLeagues(e:Event):void{
    console.log('sad')
   
  }
}
</script>
<style>
#nav {
  position: fixed;
  display: flex;
  flex-direction: row;
  align-items: center;
  background: #343a40;
  color: #fff;
  width: 100%;
  height: 80px;
}
#nav > h1 {
  font-size: 30px;
  margin-left: 20px;
}
#nav > ul {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-left: auto;
  height: 30px;
  align-content: center;
}
#nav > ul > li {
  display: flex;
  padding: 15px;
  justify-content: center;
  align-content: center;
  color: #fff;
}
#nav > ul > li :active {
  color: aquamarine;
}
#nav a {
  font-weight: bold;
  color: #fff;
  text-decoration: none;
}

#nav a.router-link-exact-active {
  color: #42b983;
}
</style>
