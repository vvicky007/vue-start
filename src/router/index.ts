import Vue from "vue";
import VueRouter, { RouteConfig } from "vue-router";
import Home from "../views/Home.vue";
import Form from "@/views/Form.vue";
Vue.use(VueRouter);
const routes: Array<RouteConfig> = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/form",
    name: "Form",
    component: Form,
  },
  {
    path: "/leagues",
    name: "Leagues",
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    components: {
      default: () =>
        import(/* webpackChunkName: "about" */ "../views/Leagues.vue"),
    },
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});
export { routes };
export default router;
