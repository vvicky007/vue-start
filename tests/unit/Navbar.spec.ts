import { shallowMount, mount, createLocalVue, Wrapper } from "@vue/test-utils";
import Navbar from "@/components/Navbar.vue";
import VueRouter from "vue-router";
import { routes } from "@/router/index";
const localVue = createLocalVue();
localVue.use(VueRouter);
describe("Navbar.vue", () => {
  let router, wrapper: Wrapper<Navbar>;
  beforeEach(() => {
    router = new VueRouter({ routes, mode: "history" });
    wrapper = mount(Navbar, {
      router,
    });
  });
  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Should display proper header", () => {
    const home = wrapper.find("h1");
    expect(home.text()).toBe("Feel Football");
  });
  it("Should display proper link names", () => {
    const links = wrapper.findAll("a");
    expect(links.length).toBe(3);
    expect(links.at(0).text()).toBe("Home");
    expect(links.at(1).text()).toBe("Leagues");
    expect(links.at(2).text()).toBe("Signin");
  });
  it("Links should contain proper links and show active links", async () => {
    const links = wrapper.findAll("a");
    await links.at(0).trigger("click");
    await wrapper.vm.$nextTick();
    expect(links.at(0).attributes().href).toEqual("/");
    expect(links.at(0).attributes().class).toContain(
      "router-link-exact-active"
    );
    expect(links.at(0).attributes().class).toContain("router-link-active");
    await links.at(1).trigger("click");
    await wrapper.vm.$nextTick();
    expect(links.at(1).attributes().href).toEqual("/leagues");
    expect(links.at(1).attributes().class).toContain(
      "router-link-exact-active"
    );
    expect(links.at(1).attributes().class).toContain("router-link-active");
    expect(wrapper.vm.$router.currentRoute.name).toEqual("Leagues");
    await links.at(2).trigger("click");
    await wrapper.vm.$nextTick();
    expect(links.at(2).attributes().href).toEqual("/form");
    expect(links.at(2).attributes().class).toContain(
      "router-link-exact-active"
    );
    expect(links.at(2).attributes().class).toContain("router-link-active");
    expect(wrapper.vm.$router.currentRoute.name).toEqual("Form");
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
