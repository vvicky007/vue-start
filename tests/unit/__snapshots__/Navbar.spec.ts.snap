// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue should match the snapshot 1`] = `
<header>
  <div>
    <div class="header"><span vif="open">Collapse</span></div>
    <nav id="nav">
      <h1>Feel Football</h1>
      <ul class="nav-list-items">
        <li><a href="/" aria-current="page" class="router-link-exact-active router-link-active">Home</a></li>
        <li><a href="/leagues" class="">Leagues</a></li>
        <li><a href="/form" class="">Signin</a></li>
      </ul>
    </nav>
  </div>
</header>
`;
